### PR TITLE
Use smarter cache cleanup algorithm

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
hopefully addresses https://github.com/appium/appium/issues/11350

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
@imurchie Doi you know why Travis tests are not executed on the PR?

**Comment by imurchie** [comment]
@mykola-mokhnach Yes. If the config is broken, Travis silently does not create a job. So nothing comes up on the PR. In this case, https://github.com/appium/appium-xcuitest-driver/commit/275eb73cfab0334734ddf6c2f838de6b786cf9f5 broke the yaml. The following change should fix it:
```diff
diff --git a/.travis.yml b/.travis.yml
index 5aa3050..875504b 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -138,9 +138,7 @@ install:
   - node --version
   - if [ "$TRAVIS_OS_NAME" = "osx" ]; then
       npm install;
-
-      # for mjpeg stuff
-      - npm install --no-save mjpeg-consumer
+      npm install --no-save mjpeg-consumer;
     else
       npm install --production;
       npm install --no-save gulp appium-gulp-plugins chai chai-as-promised chai-subset wd unzip mocha mocha-parallel-tests sync-request;
```

**Comment by mykola-mokhnach** [comment]
Thanks for the hint @imurchie . Will fix it

File: .travis.yml [hunk 1]
```diff
@@ -140,7 +140,7 @@ install:
       npm install;
 
       # for mjpeg stuff
```
Comment (imurchie) [inline .travis.yml]: This line cannot be here.

Comment (jlipps) [reply .travis.yml]: oh sorry! for some reason i thought when i did this the build ran and failed with an unrelated error. @mykola-mokhnach do you want to fix this in this PR or have me fix it independently?

Comment (mykola-mokhnach) [reply .travis.yml]: nvm, I'll try to fix it, because it is anyway necessary to check my changes

**Pull request merged**